src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -395,6 +395,7 @@ private function registerOAuthConfiguration(ContainerBuilder $container, array $
         $container->setParameter('api_platform.oauth.enabled', $this->isConfigEnabled($container, $config['oauth']));
         $container->setParameter('api_platform.oauth.clientId', $config['oauth']['clientId']);
         $container->setParameter('api_platform.oauth.clientSecret', $config['oauth']['clientSecret']);
+        $container->setParameter('api_platform.oauth.pkce', $config['oauth']['pkce']);
```

Comment:
IIRC using parameters like this is a deprecated practice since Symfony 3. You can just set the value extracted from the service as a service argument.

Reply:
I thought it too, but wanted to be consistent with the other parameters...
Shouldn't we deprecate the other parameters and migrate them to direct injection in 3.0?

Reply:
We can do that yes, but in another PR.